src/tools/compiletest/src/runtest.rs

```diff
@@ -674,6 +676,21 @@ impl<'test> TestCx<'test> {
         Ok(joined_lines)
     }
 
+    /// Replace line numbers in coverage reports with the placeholder `LL`,
+    /// so that the tests are less sensitive to lines being added/removed.
+    fn anonymize_coverage_line_numbers(coverage: &str) -> Cow<'_, str> {
+        // The coverage reporter prints line numbers at the start of a line.
+        // They are truncated or left-padded to occupy exactly 5 columns.
+        // (`LineNumberColumnWidth` in `SourceCoverageViewText.cpp`.)
+        // A pipe character `|` appears immediately after the final digit.
+        //
+        // Line numbers that appear inside expansion/instantiation subviews
+        // have an additional prefix of `  |` for each nesting level.
+        static LINE_NUMBER_RE: Lazy<Regex> =
+            Lazy::new(|| Regex::new(r"(?m)^(?<prefix>(?:  \|)*)[ 0-9]{4}[0-9]\|").unwrap());
```

Comment:
This regex is slightly imprecise, because `[ 0-9]{4}[0-9]` will also match things like `0 0 0`. But since the reporter is unlikely to ever emit such weird output, I decided that a more precise regex wouldn't be worth the extra complexity.

(We'd basically have to write `(    [0-9])|(   [0-9}{2})|(  [0-9]{3})|( [0-9]{4})|([0-9]{5})`, which is rather unpleasant.)